src/components/posts/NewPostButton.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <v-dialog 
+    v-model="dialog"
+    max-width="800"
+    persistent
+  >
+    <template v-slot:activator="{on, attrs}">
+      <v-fab-transition>
+        <v-btn
+          v-bind="attrs"
+          class="new_post_button"
+          color="primary"
+          dark
+          block
+          v-on="on"
+        >
+          <v-icon>mdi-pencil</v-icon>
+          Add new post
+        </v-btn>
+      </v-fab-transition>
+    </template>
+
+    <v-card>
+      <v-card-title>
+        New Post
+      </v-card-title>
+      <v-card-text>
+        <v-form
+          ref="form"
+          @submit.prevent="addPost"
+        >
+          <v-textarea
+            v-model="post.newPostContent"
+            outlined
+            placeholder="Write something..."
+            clearable
+            counter
+            rows="4"
+          />
+          <v-card-actions class="justify-space-between py-3">
+            <v-btn
+              @click="cancel"
+            >
+              CANCEL
```

Comment:
use i18n as above

```suggestion
                      {{ $t('general.cancel') }}
```

use key form **general** group